compiler/rustc_monomorphize/src/collector.rs

```diff
@@ -1207,6 +1210,15 @@ fn collect_items_of_instance<'tcx>(
     mentioned_items: &mut MonoItems<'tcx>,
     mode: CollectionMode,
 ) {
+    // Check the instance for feature-dependent ABI if we cannot determine that the function has
+    // "Rust" ABI, which is known not to be feature-dependent.
+    // Note that the check is still correct on Rust ABI functions, but somewhat expensive. Hence,
+    // checking for "Rust" ABI is just an optimization.
+    let instance_ty = instance.ty(tcx, ParamEnv::reveal_all());
+    if !instance_ty.is_fn() || !instance_ty.fn_sig(tcx).abi().is_rust() {
```

Comment:
What's the point of `!instance_ty.is_fn()`?

Reply:
Added a comment.